src/models/PrimaryAuth.js

```diff
@@ -37,9 +37,24 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
       }
 
       return {
-        username: ['string', true, username],
+        username: {
+          type: 'string',
+          validate: function (value) {
+            if(!value || value.length < 1) {
+              return 'Please enter a username';
```

Comment:
![errormessage mov](https://cloud.githubusercontent.com/assets/20933838/22994328/39c22a6a-f3cf-11e6-9705-c32bd9491172.gif)


I'll add i18n soon. 

> also since required is set to true is there a way to set an error message for that ?

I haven't found a way to do that.